File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/core/note/NoteRepository.java
package coolpharaoh.tee.speicher.tea.timer.core.note;

import android.app.Application;

import java.util.List;

import coolpharaoh.tee.speicher.tea.timer.core.database.TeaMemoryDatabase;

public class NoteRepository {

    private final NoteDao noteDao;

    public NoteRepository(Application application) {
        TeaMemoryDatabase database = TeaMemoryDatabase.getDatabaseInstance(application);
        noteDao = database.getNoteDao();
    }

    public void insertNote(Note note) {
        noteDao.insert(note);
    }

    public void updateNote(Note note) {
        noteDao.update(note);
    }

    public List<Note> getNotes() {
        return noteDao.getNotes();
    }

    public Note getNoteByTeaIdAndPosition(long teaId, int position) {
        return noteDao.getNoteByTeaIdAndPosition(teaId, position);
    }

    public List<Note> getNotesByTeaIdAndPositionBiggerZero(long teaId) {
        return noteDao.getNotesByTeaIdAndPositionBiggerZero(teaId);
    }

    public void deleteNoteByTeaIdAndPosition(long teaId, int position) {
        noteDao.deleteNoteByTeaIdAndPosition(teaId, position);
    }
}

File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/core/note/NoteRepository.kt
package coolpharaoh.tee.speicher.tea.timer.core.note

import android.app.Application
import coolpharaoh.tee.speicher.tea.timer.database.TeaMemoryDatabase

class NoteRepository(application: Application) {
    private val noteDao: NoteDao

    init {
        val database = TeaMemoryDatabase.getDatabase(application)
        noteDao = database.noteDao
    }

    fun insertNote(note: Note) {
        noteDao.insert(note)
    }

    fun updateNote(note: Note) {
        noteDao.update(note)
    }

    val notes: List<Note>
        get() = noteDao.notes

    fun getNoteByTeaIdAndPosition(teaId: Long, position: Int): Note? {
        return noteDao.getNoteByTeaIdAndPosition(teaId, position)
    }

    fun getNotesByTeaIdAndPositionBiggerZero(teaId: Long): List<Note> {
        return noteDao.getNotesByTeaIdAndPositionBiggerZero(teaId)
    }

    fun deleteNoteByTeaIdAndPosition(teaId: Long, position: Int) {
        noteDao.deleteNoteByTeaIdAndPosition(teaId, position)
    }
}